pre_commit_hooks/loaderon_hooks/general_hooks/check_branch_name.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+import re
+
+from pre_commit_hooks.loaderon_hooks.util.git_helpers import get_current_branch_name
+from pre_commit_hooks.loaderon_hooks.util.template_methods.checker_template_method import CheckerTemplateMethod
+
+
+class BranchNameChecker(CheckerTemplateMethod):
```

Comment:
while the class-based approach can "dry" up some code, I find it's much harder to reason about what's going on -- in most of the files the only "duplication" is the argument parsing, which isn't really all that much to save from classes

the base class name here also isn't that great -- I would maybe say something like `CheckerBase` or some such since it's not a "method"